pkg/api/frontend_logging.go

```diff
@@ -1,104 +1,40 @@
 package api
 
 import (
-	"fmt"
-	"strings"
-
 	"github.com/getsentry/sentry-go"
+	"github.com/grafana/grafana/pkg/api/frontendlogging"
 	"github.com/grafana/grafana/pkg/api/response"
 	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/models"
-	"github.com/inconshreveable/log15"
 )
 
 var frontendLogger = log.New("frontend")
 
-type frontendSentryExceptionValue struct {
-	Value      string            `json:"value,omitempty"`
-	Type       string            `json:"type,omitempty"`
-	Stacktrace sentry.Stacktrace `json:"stacktrace,omitempty"`
-}
-
-type frontendSentryException struct {
-	Values []frontendSentryExceptionValue `json:"values,omitempty"`
-}
-
-type frontendSentryEvent struct {
-	*sentry.Event
-	Exception *frontendSentryException `json:"exception,omitempty"`
-}
+type frontendLogMessageHandler func(c *models.ReqContext, event frontendlogging.FrontendSentryEvent) response.Response
 
-func (value *frontendSentryExceptionValue) FmtMessage() string {
-	return fmt.Sprintf("%s: %s", value.Type, value.Value)
-}
+func NewFrontendLogMessageHandler(store *frontendlogging.SourceMapStore) frontendLogMessageHandler {
```

Comment:
Is this part kept in the `api` package to avoid cyclic imports? Otherwise it would be neat to move this as well to `api/frontendlogging` so `api` can be slimmed down :)

Reply:
yup, exactly. Test utils for http handlers are in `api`. In such case they should be extracted as well, but felt like too big a scope for this PR :)

Reply:
OK, agree then :)